src/transformStateWithClones.js

```diff
@@ -6,8 +6,37 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentStateCopy = { ...state };
+
+  for (const ch of actions) {
+    const stateCopy = { ...currentStateCopy };
+
+    if (ch.type === 'addProperties') {
+      for (const [key, value] of Object.entries(ch.extraData)) {
+        stateCopy[key] = value;
+      }
+    }
+
+    if (ch.type === 'removeProperties') {
+      for (const key of ch.keysToRemove) {
+        delete stateCopy[key];
+      }
+    }
+
+    if (ch.type === 'clear') {
+      for (const key in stateCopy) {
+        delete stateCopy[key];
+      }
+    }
```

Comment:
it is better to use switch case in such cases

Reply:
Yes, i agree, next time I`ll try to use switch case. Thank you!